src/app/cartAtom.ts

```diff
@@ -0,0 +1,7 @@
+import { atom } from "recoil";
+import { CartItem } from "./hooks/useCart";
+
+export const cartState = atom<CartItem[]>({
+  key: "cartState",
+  default: JSON.parse(localStorage.getItem("cart") || "[]"),
```

Comment:
This will most likely cause a build error, localstorage is only available client side, add "use client" to the top of the file.